src/modules/mana/utils.ts

```diff
@@ -1,6 +1,94 @@
+import { BigNumber, ethers } from 'ethers'
+import { graphql } from 'decentraland-dapps/dist/lib/graph'
+
 export const MANA_CONTRACT_ADDRESS = process.env
   .REACT_APP_MANA_CONTRACT_ADDRESS!
 export const ERC20_PREDICATE_CONTRACT_ADDRESS = process.env
   .REACT_APP_ERC20_PREDICATE_CONTRACT_ADDRESS!
 export const ROOT_CHAIN_MANAGER_CONTRACT_ADDRESS = process.env
   .REACT_APP_ROOT_CHAIN_MANAGER_CONTRACT_ADDRESS!
+export const MATIC_ROOT_CHAIN_SUBGRAPH = process.env
+  .REACT_APP_MATIC_ROOT_CHAIN_SUBGRAPH!
+
+const POLL_INTERVAL = 30 * 1000 // 30 seconds
+
+function instantiateStateReceiver() {
+  const provider = new ethers.providers.JsonRpcProvider(
+    'https://rpc-mumbai.matic.today'
+  )
+  return new ethers.Contract(
+    '0x0000000000000000000000000000000000001001',
+    [
+      {
+        constant: true,
+        inputs: [],
+        name: 'lastStateId',
+        outputs: [
+          {
+            internalType: 'uint256',
+            name: '',
+            type: 'uint256',
+          },
+        ],
+        payable: false,
+        stateMutability: 'view',
+        type: 'function',
+      },
+    ] as any,
+    provider
+  )
+}
+
+export async function isWithdrawalSynced(txHash: string) {
+  const provider = new ethers.providers.JsonRpcProvider(
+    'https://rpc-mumbai.matic.today'
```

Comment:
Should we need to move this to an env or get it from `decentraland-dapps` or `decentraland-transactions`?

Reply:
ahh sorry! yes this shouldn't be hardcoded haha